hooks/helmlint.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# We don't set `set -e` in this script because some of the functions will return a non-zero value (e.g contains_element)
+
+# OSX GUI apps do not pick up environment variables the same way as Terminal apps and there are no easy solutions,
+# especially as Apple changes the GUI app behavior every release (see https://stackoverflow.com/q/135688/483528). As a
+# workaround to allow GitHub Desktop to work, add this (hopefully harmless) setting here.
+export PATH=$PATH:/usr/local/bin
+
+# Take the current working directory to know when to stop walking up the tree
+readonly cwd_abspath="$(realpath "$PWD")"
+
+# An array to keep track of which charts we already linted
+seen_chart_paths=()
+
+# https://stackoverflow.com/a/8574392
+# Usage: contains_element "val" "${array[@]}"
+# Returns: 0 if there is a match, 1 otherwise
+contains_element() {
+  local -r match="$1"
+  shift
+
+  for e in "$@"; do
+    if [[ "$e" == "$match" ]]; then
+      return 0
+    fi
+  done
+  return 1
+}
+
+# Only log debug statements if PRECOMMIT_DEBUG environment variable is set.
+debug() {
+  if [[ $PRECOMMIT_DEBUG ]]; then
+    echo "$@"
+  fi
+}
+
+# Recursively walk up the tree until the current working directory and check if the changed file is part of a helm
+# chart. Helm charts have a Chart.yaml file.
+# Return value is stored in `chart_path_return`
+chart_path_return=""
+chart_path() {
+  # Return return value
+  chart_path_return=""
+
+  # We check both the current dir as well as the parent dir, in case the current dir is a file
+  local -r changed_file="$(realpath "$1")"
```

Comment:
Read `$1` into a named var above to make it easier to see what params this function takes.